### ISSUE TITLE
Error: Plugin did not respond while importing

### ISSUE THREAD

**Issue by soniadas0210** [hashicorp/terraform-provider-azurerm#19065]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Community Note

* Please vote on this issue by adding a :thumbsup: [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform Version

Terraform v1.3.3

### AzureRM Provider Version

3.29.0

### Affected Resource(s)/Data Source(s)

azurerm_kubernetes_cluster.aks

### Terraform Configuration Files

```hcl
resource "azurerm_kubernetes_cluster" "aks" {
  automatic_channel_upgrade         = "node-image"
  azure_policy_enabled              = var.policy_add_on_enabled
  dns_prefix                        = var.dns_prefix
  enable_pod_security_policy        = false
  http_application_routing_enabled  = false
  kubernetes_version                = var.kubernetes_version
  local_account_disabled            = false
  location                          = var.location
  name                              = var.cluster_name
  open_service_mesh_enabled         = false
  public_network_access_enabled     = true
  resource_group_name               = var.resource_group_name
  role_based_access_control_enabled = true
  sku_tier                          = "Free"
  tags = tomap(
    {
      Environment = "${var.aks_tag_environment}"
      Region      = "${var.aks_tag_region}"
    }
  )

  azure_active_directory_role_based_access_control {
    admin_group_object_ids = [
      "*********-**********-*************",
      "*********-**********-*************",
    ]
    azure_rbac_enabled = false
    managed            = true
    tenant_id          = "*********-**********-*************"
  }

  default_node_pool {
    enable_auto_scaling   = true
    enable_node_public_ip = false
    kubelet_disk_type     = "OS"
    max_count             = var.aks_default_node_pool_max_count
    max_pods              = 110
    min_count             = var.aks_default_node_pool_min_count
    name                  = "core"
    node_labels           = {}
    node_taints           = []
    os_disk_size_gb       = 128
    os_disk_type          = "Managed"
    os_sku                = "Ubuntu"
    tags                  = {}
    type                  = "VirtualMachineScaleSets"
    vm_size               = "Standard_B2ms"
    zones                 = var.aks_default_node_pool_zones
    orchestrator_version  = var.node_pool_kubernetes_orchestrator_version
  }

  identity {
    identity_ids = []
    type         = "SystemAssigned"
  }

  ingress_application_gateway {
    gateway_name = var.application_gateway_name
    subnet_cidr  = "XX.X.X.X/XX"
  }

  linux_profile {
    admin_username = "***********"

    ssh_key {
      key_data = <<-EOT
                ssh-rsa ********************************************
            EOT
    }
  }

  network_profile {
    dns_service_ip     = "XX.XX.XX.XX"
    docker_bridge_cidr = "XXX.XX.XX.XX/XX"
    load_balancer_sku  = "******"
    network_plugin     = "*****"
    outbound_type      = "loadBalancer"
    pod_cidr           = "XX.XXX.XX.XX/XX"
    service_cidr       = "XX.X.X.X/XX"
  }
}
```

### Debug Output/Panic Output

```shell
Stack trace from the terraform-provider-azurerm_v3.29.0_x5 plugin:

panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x2 addr=0x0 pc=0x10704b09c]

goroutine 26 [running]:
github.com/hashicorp/terraform-provider-azurerm/internal/services/containers.resourceKubernetesClusterRead(0x1400180b300, {0x1086794c0?, 0x140006ee000})
	github.com/hashicorp/terraform-provider-azurerm/internal/services/containers/kubernetes_cluster_resource.go:1885 +0x66c
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).read(0x10909ba88?, {0x10909ba88?, 0x14000ae57d0?}, 0xd?, {0x1086794c0?, 0x140006ee000?})
	github.com/hashicorp/terraform-plugin-sdk/v2@v2.18.0/helper/schema/resource.go:712 +0x138
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).RefreshWithoutUpgrade(0x1400087d5e0, {0x10909ba88, 0x14000ae57d0}, 0x14001a9e820, {0x1086794c0, 0x140006ee000})
	github.com/hashicorp/terraform-plugin-sdk/v2@v2.18.0/helper/schema/resource.go:1015 +0x46c
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*GRPCProviderServer).ReadResource(0x14000289b18, {0x10909ba88?, 0x14000ae5710?}, 0x1400124e440)
	github.com/hashicorp/terraform-plugin-sdk/v2@v2.18.0/helper/schema/grpc_provider.go:613 +0x40c
github.com/hashicorp/terraform-plugin-go/tfprotov5/tf5server.(*server).ReadResource(0x14000c02460, {0x10909ba88?, 0x140002f31d0?}, 0x14001a820c0)
	github.com/hashicorp/terraform-plugin-go@v0.10.0/tfprotov5/tf5server/server.go:747 +0x39c
github.com/hashicorp/terraform-plugin-go/tfprotov5/internal/tfplugin5._Provider_ReadResource_Handler({0x108cadb00?, 0x14000c02460}, {0x10909ba88, 0x140002f31d0}, 0x14001a82060, 0x0)
	github.com/hashicorp/terraform-plugin-go@v0.10.0/tfprotov5/internal/tfplugin5/tfplugin5_grpc.pb.go:349 +0x174
google.golang.org/grpc.(*Server).processUnaryRPC(0x1400013aa80, {0x1090adc18, 0x140006c2680}, 0x14001a84000, 0x140016af290, 0x10d1b2db0, 0x0)
	google.golang.org/grpc@v1.47.0/server.go:1283 +0xb64
google.golang.org/grpc.(*Server).handleStream(0x1400013aa80, {0x1090adc18, 0x140006c2680}, 0x14001a84000, 0x0)
	google.golang.org/grpc@v1.47.0/server.go:1620 +0x840
google.golang.org/grpc.(*Server).serveStreams.func1.2()
	google.golang.org/grpc@v1.47.0/server.go:922 +0x88
created by google.golang.org/grpc.(*Server).serveStreams.func1
	google.golang.org/grpc@v1.47.0/server.go:920 +0x298

Error: The terraform-provider-azurerm_v3.29.0_x5 plugin crashed!
```

### Steps to Reproduce

terraform import <resource_name> <azure_resource_id>

**Comment by tombuildsstuff** [comment]
hey @soniadas0210 

Thanks for opening this issue.

Taking a look through here it appears that this crash has been fixed in `v3.29.1` of the Azure Provider which was released on Friday, if you can update to that version then this should import correctly.

Since this can be fixed by updating the version of the Azure Provider being used, I'm going to close this issue for the moment - but if you're still encountering this issue in `v3.29.1` then please feel free to open a new issue and we can take another look.

Thanks!

**Comment by soniadas0210** [comment]
Thank You. it is working now.
